### ISSUE TITLE
questions about the three data files

### ISSUE THREAD

**Issue by CaLlMeErIC** [junjieliang672/FARE#1]
Hi, I was recently trying to reproduce the method mentioned in your paper 'FARE: Enabling Fine-grained Attack Categorization
under Low-quality Labeled Data’. I found some problems that I can't understand

1. What is the relationship between the three data files 'malware.npy', 'malware_test.mat', and 'malware_train.mat' in the data folder? I found that both malware_test.mat and malware_train.mat have different dimensions, are they generated by malware.npy?

2. I have collected some malware report samples myself, but don't know how to convert them into arrays and vectors that are fed into the neural network. Can you share this part of the code that extracts the features from the malware report?

3. I read in the paper that the feature dimension of malware is 100, but I found that the dimension of 'malware.npy' is 189000*64, how did the 64 dimension here come from?

In addition, I found that Fare.py has a small error with redundant parameters in line 92. After modifying it to testX, testY = getTest(data_dic['flag']), it can run normally

It would be very helpful if you could help me answer these questions, and I wish you the best of luck in your work and more progress in your research

**Comment by junjieliang672** [comment]
Hi Eric,

Thank you for your interest in our paper. We have a detailed description of the data files and codes in the Data Format section in ["readme.md".](https://github.com/junjieliang672/FARE/blob/master/readme.md). You can find answers to Q1 and Q3 there. As for Q2, unfortunately, there is no universal way to extract features from a report since the format of different reports varies dramatically from case to case. As such I'm afraid I can't answer your question here. If you are confused about how to format the feature and label to a valid training and testing file, please refer to our code ["readData.py"](https://github.com/junjieliang672/FARE/blob/master/readData.py).

As for the bug in line 92 of ["FARE.py"](https://github.com/junjieliang672/FARE/blob/master/FARE.py), we really appreciate your report. We have removed the redundant parameters accordingly.

Hope our answer help solve your questions.

Best,
Junjie

**Comment by CaLlMeErIC** [comment]
Thanks for your replies！My understanding is that the 64 dimensions in malware.npy correspond to the given label and the labels generated by 63 unsupervised classification methods. Is this understanding correct? If so can you provide details on these unsupervised classification methods, or share the code details for this part?
